### DISCUSSION TITLE
Why do we have to inject jQuery just to have the test pass?

### DISCUSSION THREAD

**Discussion by astridx** [joomla/joomla-cms#32766] [Q&A] [answered]
I have read 

https://github.com/joomla/joomla-cms/pull/32707

and

https://github.com/joomla/joomla-cms/pull/32740#issuecomment-802313870

Therefore I suspect that it is about this line:
https://github.com/joomla/joomla-cms/blob/83a61f1edd39d365185d397e1a88cdd2f9cb81e1/administrator/components/com_content/tmpl/articles/default.php#L30

I commented out this line and ran the unit tests and the codeception tests for com_content. 
No error occurred. 

[unit_tests.txt](https://github.com/joomla/joomla-cms/files/6176443/unit_tests.txt)
[com_content.txt](https://github.com/joomla/joomla-cms/files/6176444/com_content.txt)

**Comment by Fedik** [comment, marked as answer]
Please make a PR for removing it, then we will see :wink:

**Reply by brianteeman** [reply]
see #32707

**Reply by dgrammatiko** [reply]
The answer is simple and it's because of this: https://github.com/joomla-projects/joomla-browser/blob/f997dce1100ef00a0b83ecf09755df37bf419b0f/src/JoomlaBrowser.php#L693-L702

**Reply by Fedik** [reply]
I just wanted to post same link :smile:

**Reply by Fedik** [reply]
yea it a issue with `joomla-projects/joomla-browser`

**Reply by dgrammatiko** [reply]
BTW jQuery was injected just to have passing tests in `administrator/components/com_content/tmpl/articles/default.php` and  `administrator/components/com_fields/tmpl/fields/default.php ` according to https://github.com/joomla/joomla-cms/pull/31990

**Reply by Fedik** [reply]
well, I have opened the issue https://github.com/joomla-projects/joomla-browser/issues/215

**Reply by dgrammatiko** [reply]
I think the function should be duplicated with another name and without jQuery and use in the 4.0 branch but the original needs to stay untouched for the v3 branch. Unless if the vanilla code could work on both branches. BTW it's codeception that fails not unit tests

**Reply by astridx** [reply]
Thank you. Now I understand. In my tests, this was not problematic. But in other places it throws errors for me too.

**Comment by astridx** [comment]
I have just checked it. It's not true that we no longer use jQuery in Joomla 4. I just found a place here.
https://github.com/joomla/joomla-cms/blob/c78ef2fb45f68c91d87aa553532dceb4d66478db/build/media_source/system/js/searchtools.es6.js#L253

**Reply by dgrammatiko** [reply]
That's a special case and it goes like `if jQuery is loaded and also chosen is loaded`, so we are not using jQuery we just cover also that case.

Actually this probably doesn't even work and honestly Joomla 4 **should not ship** dead code (chosen is dead for some years now)
